src/tests/test_yaraparse.py

```diff
@@ -24,3 +23,115 @@ def test_literal_to_hex():
     result = ursify_plain_string(ascii_str.pure_text, is_ascii=True)
 
     assert result.query == "{616263}"
+
+
+def rule_to_query(rule):
+    result = parse_yara(rule)
+    (rule,) = result
+    parsed = rule.parse()
+    return parsed.query
+
+
+def test_condition_gt():
+    query = rule_to_query(
+        """
+    rule test {
+        strings:
+            $x = "test"
+        condition:
+            #x > 1
+    }"""
+    )
+    assert query == "{74657374}"
+
+
+def test_condition_lt():
+    query = rule_to_query(
+        """
+    rule test {
+        strings:
+            $x = "test"
+        condition:
+            1 < #x
+    }"""
+    )
+    assert query == "{74657374}"
+
+
+def test_condition_ge():
+    query = rule_to_query(
+        """
+    rule test {
+        strings:
+            $x = "test"
+        condition:
+            #x >= 1
+    }"""
+    )
+    assert query == "{74657374}"
+
+
+def test_condition_gt_reversed():
+    query = rule_to_query(
+        """
+    rule test {
+        strings:
+            $x = "test"
+        condition:
+            1 < #x
+    }"""
+    )
+    assert query == "{74657374}"
+
+
+def test_condition_eq():
+    query = rule_to_query(
+        """
+    rule test {
+        strings:
+            $x = "test"
+        condition:
+            #x == 1
+    }"""
+    )
+    assert query == "{74657374}"
+
+
+def test_condition_eq_rev():
+    query = rule_to_query(
+        """
+    rule test {
+        strings:
+            $x = "test"
+        condition:
+            1 == #x
+    }"""
+    )
+    assert query == "{74657374}"
+
+
+def test_condition_eq0():
+    query = rule_to_query(
+        """
+    rule test {
+        strings:
+            $x = "test"
+        condition:
+            #x == 0
+    }"""
+    )
+    assert query == "{74657374}"
```

Comment:
Shouldn't this eval to `{}`?

Reply:
Of course. It started failing after fixing the bug above. Thanks for noticing.